### ISSUE TITLE
step ambiguity issue

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber/cucumber#217]
_From @tony-kerz on June 6, 2017 15:44_

the following step definitions:
```
    When('we HTTP GET "{path}"', async function(path) {
      //...
    })

    When('we HTTP GET "{path}" with query "{query}"', async function(path, query) {
      //...
    })
```
incurs the following: 
```
1) Scenario: index with query - test/features/test.feature:23
   Step: When we HTTP GET "/" with query "_id=2" - test/features/test.feature:24
   Message:
     Multiple step definitions match:
       we HTTP GET "{path}"                      - src/index.js:41
       we HTTP GET "{path}" with query "{query}" - src/index.js:45
```
should there be implied begin-end terminators in these expressions?

_Copied from original issue: cucumber/cucumber-expressions-javascript#5_

**Comment by charlierudolph** [comment]
To me, this requires you somehow to specify that path should not contain a double quote otherwise the resulting regex's will cause it to be ambiguous. cucumber-js has the built in transform for strings with in [double quotes](https://github.com/cucumber/cucumber-js/blob/deffdcf6ac35c074bd3914c6c3dc75ca95ebfa1a/docs/support_files/api_reference.md#defineparametertyperegexp-typename-transformer)

**Comment by aslakhellesoy** [comment]
This should no longer be an issue when #190 and #191 are fixed

**Comment by aslakhellesoy** [comment]
Should be fixed in v4.0.0. Please reopen a new ticket referencing this one.
